test/Microsoft.ML.Benchmarks/Harness/Metrics.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.ML.Benchmarks
+{
+    public abstract class WithExtraMetrics
+    {
+        protected abstract IEnumerable<Metric> GetMetrics();
+
+        /// <summary>
+        ///  this method is executed after running the benchmrks
+        ///  we use it as hack to simply print to console so ExtraMetricColumn can parse the output
+        /// </summary>
+        [GlobalCleanup]
+        public void ReportMetrics()
+        {
+            foreach (var metric in GetMetrics())
```

Comment:
nit: Would it improve perf to set `var metrics = GetMetrics();` right before the foreach loop and then write the condition as `var metric in metrics`? Not sure...

Reply:
@briancylui no, it would not.

Whenever you are not sure about something you can benchmark it with BenchmarkDotNet ;)